src/rgw/rgw_rados.cc

```diff
@@ -8408,8 +8408,13 @@ int RGWRados::delete_system_obj(rgw_raw_obj& obj, RGWObjVersionTracker *objv_tra
   return 0;
 }
 
-int RGWRados::delete_obj_index(const rgw_obj& obj)
+int RGWRados::delete_obj_index(const rgw_obj& _obj)
 {
+  rgw_obj obj = _obj;
+  if (obj.key.instance == "null") {
```

Comment:
@oritwas  I clear the instance in rgw_remove_object now and delete_obj_index stays the name.